uicomponent-compose/core/src/main/java/io/github/droidkaigi/feeder/core/ScrollableTabRow.kt

```diff
@@ -0,0 +1,379 @@
+) {
+    Surface(
+        modifier = modifier.selectableGroup(),
+        color = backgroundColor,
+        contentColor = contentColor
+    ) {
+        val scrollState = rememberScrollState()
+        val coroutineScope = rememberCoroutineScope()
+        val scrollableTabData = remember(scrollState) {
+            ScrollableTabData(
+                scrollState = scrollState,
+                selectedTab = selectedTabIndex,
+                coroutineScope = coroutineScope
+            )
+        }
+        SubcomposeLayout(
+            Modifier
+                .fillMaxWidth()
+                .wrapContentSize(align = Alignment.CenterStart)
+                .horizontalScroll(scrollState)
+                .clipToBounds()
+        ) { constraints ->
+            val minTabWidth = ScrollableTabRowMinimumTabWidth.roundToPx()
+            val padding = edgePadding.roundToPx()
+            val tabConstraints = constraints.copy(minWidth = minTabWidth)
+
+            val tabPlaceables = subcompose(TabSlots.Tabs, tabs)
+                .fastMap { it.measure(tabConstraints) }
+
+            var layoutWidth = padding * 2
+            var layoutHeight = 0
+            tabPlaceables.fastForEach {
+                layoutWidth += it.width
+                layoutHeight = maxOf(layoutHeight, it.height)
+            }
+
+            // Position the children.
+            layout(layoutWidth, layoutHeight) {
+                // Place the tabs
+                val tabPositions = mutableListOf<TabPosition>()
+                var left = padding
+                tabPlaceables.fastForEach {
+                    tabPositions.add(TabPosition(left = left.toDp(), width = it.width.toDp()))
+                    left += it.width
+                }
+
+                // The divider is measured with its own height, and width equal to the total width
+                // of the tab row, and then placed on top of the tabs.
+                subcompose(TabSlots.Divider, divider).fastForEach {
+                    val placeable = it.measure(
+                        constraints.copy(minWidth = layoutWidth, maxWidth = layoutWidth)
+                    )
+                    placeable.placeRelative(0, layoutHeight - placeable.height)
+                }
+
+                subcompose(TabSlots.BackgroundIndicator) {
+                    backgroundIndicator(tabPositions)
+                }.fastForEach {
+                    it.measure(Constraints.fixed(layoutWidth, layoutHeight)).placeRelative(0, 0)
+                }
+
+                left = padding
+                tabPlaceables.fastForEach {
+                    it.placeRelative(left, 0)
+                    left += it.width
+                }
+
+                subcompose(TabSlots.ForegroundIndicator) {
+                    foregroundIndicator(tabPositions)
+                }.fastForEach {
+                    it.measure(Constraints.fixed(layoutWidth, layoutHeight)).placeRelative(0, 0)
+                }
```

Comment:
In the original ScrollTabRow, the indicator is drawn after the text, so it will be displayed above.
This custom ScrollTabRow allows to display the Indicator behind the text.
https://github.com/androidx/androidx/blob/master/compose/material/material/src/commonMain/kotlin/androidx/compose/material/TabRow.kt#L258-L291

Reply:
👍

Reply:
Super cool!! I'm learning a lot. 🆒